pkg/controller/istiocsr/services.go

```diff
@@ -0,0 +1,87 @@
+package istiocsr
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/openshift/cert-manager-operator/api/operator/v1alpha1"
+	"github.com/openshift/cert-manager-operator/pkg/operator/assets"
+)
+
+const (
+	// grpcServicePortName is the name found for the GRPC service in the static manifest.
+	grpcServicePortName = "web"
+)
+
+func (r *Reconciler) createOrApplyServices(istiocsr *v1alpha1.IstioCSR, resourceLabels map[string]string, istioCSRCreateRecon bool) error {
+	desired := r.getServiceObject(istiocsr, resourceLabels)
+
+	serviceName := fmt.Sprintf("%s/%s", desired.GetNamespace(), desired.GetName())
+	r.log.V(1).Info("reconciling service resource", "name", serviceName)
+	fetched := &corev1.Service{}
+	key := types.NamespacedName{
+		Name:      desired.GetName(),
+		Namespace: desired.GetNamespace(),
+	}
+	exist, err := r.Exists(r.ctx, key, fetched)
+	if err != nil {
+		return FromClientError(err, "failed to check %s service resource already exists", serviceName)
+	}
+
+	if exist && istioCSRCreateRecon {
+		r.eventRecorder.Eventf(istiocsr, corev1.EventTypeWarning, "ResourceAlreadyExists", "%s service resource already exists, maybe from previous installation", serviceName)
+	}
+	if exist && hasObjectChanged(desired, fetched) {
+		r.log.V(1).Info("service has been modified, updating to desired state", "name", serviceName)
```

Comment:
The event recorders will indicate this flow.

Reply:
suggestion incorporated. Logs levels updated.